### PR TITLE
DMC: Verrak, Warped Sengir and support

### PR DESCRIPTION
**Pull request by Northmoc**
not sure if the line in GameActionUtil is necessary...

### CODE REVIEW

File: forge-game/src/main/java/forge/game/trigger/TriggerHandler.java [hunk 1]
```diff
@@ -564,6 +564,15 @@ private void runSingleTriggerInternal(final Trigger regtrig, final Map<AbilityKe
             Card triggeredCard = ((Card) sa.getTriggeringObject(AbilityKey.Card));
             host.addRemembered(triggeredCard);
         }
+        if (regtrig.hasParam("LifePaid")) {
+            final SpellAbility trigSA = (SpellAbility) runParams.get(AbilityKey.CastSA);
+            int amount = trigSA.getAmountLifePaid();
+            if (amount > 0) {
+                host.addRemembered(amount);
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/trigger/TriggerHandler.java:571]: Remembering it on the host would probably break if you have such ability on the stack twice, one with life paid and one without?

Comment (Northmoc) [reply forge-game/src/main/java/forge/game/trigger/TriggerHandler.java:571]: Good call, different amounts might also get messy. Hmm, back to the drawing board – any ideas?

Comment (tool4ever) [reply forge-game/src/main/java/forge/game/trigger/TriggerHandler.java:571]: Imo the CastSA should persist that info, so you should query TriggeringObjects directly

Comment (Northmoc) [reply forge-game/src/main/java/forge/game/trigger/TriggerHandler.java]: Good instinct on this – I did test it quickly with the old code and it definitely didn't handle multiple triggers well. This new version seems to handle this case correctly

File: forge-game/src/main/java/forge/game/trigger/TriggerHandler.java [hunk 2]
```diff
@@ -564,6 +564,12 @@ private void runSingleTriggerInternal(final Trigger regtrig, final Map<AbilityKe
             Card triggeredCard = ((Card) sa.getTriggeringObject(AbilityKey.Card));
             host.addRemembered(triggeredCard);
         }
+        if (regtrig.hasParam("LifePaid")) {
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/trigger/TriggerHandler.java:567]: Looks cleaner already :)

I think this check probably doesn't belong here, it's also a bit late and can cause more engine calls than needed
`meetsRequirementsOnTriggeredObjects` should be a better place for it

Comment (Northmoc) [reply forge-game/src/main/java/forge/game/trigger/TriggerHandler.java:567]: Thanks – I'll try moving this

**Review by tool4ever — APPROVED** [review]

File: forge-game/src/main/java/forge/game/player/Player.java [hunk 3]
```diff
@@ -629,6 +629,7 @@ public final boolean payLife(final int lifePayment, final SpellAbility cause, fi
         }
 
         loseLife(lifePayment, false, false);
+        cause.setPaidLife(lifePayment);
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/player/Player.java:632]: i don't know if this is the right part for this?

shouldn't that only be set for when life is paid as cost?

Comment (kevlahnota) [reply forge-game/src/main/java/forge/game/player/Player.java:632]: gonna commit this, just update this/open an issue later on if needed be.

**Pull request merged**